src/service.ts

```diff
@@ -284,10 +327,25 @@ export class DebugService implements IDebugger {
       (breakpoint, i, arr) =>
         arr.findIndex(el => el.line === breakpoint.line) === i
     );
-    this._model.breakpointsModel.breakpoints = kernelBreakpoints;
+    this._model.breakpointsModel.setBreakpoints(code, kernelBreakpoints);
```

Comment:
Should that be `dumpedCell.sourcePath` instead?

```suggestion
    this._model.breakpointsModel.setBreakpoints(dumpedCell.sourcePath, kernelBreakpoints);
```

Reply:
Or calculating the cell id with `this._debuggerService.getCellId(cell.model.value.text)`

Reply:
:sweat_smile: